File: c/rna-transcription/src/rna_transcription.c

#include "rna_transcription.h"


char *to_rna(const char *dna){

    int i = 0;
    int string_len = strlen(dna);

    char *output = malloc(sizeof(char) * string_len);

    for(i = 0; i < string_len; i++)
    {
        switch(dna[i]){
            case('G'):
                // C
                output[i] = 'C';
                break;
            case('C'):
                // G
                output[i] = 'G';
                break;
            case('T'):
                // A
                output[i] = 'A';
                break;
            case('A'):
                // U
                output[i] = 'U';
                break;
        }
    }


    return output;
}

File: go/rna-transcription/rna_transcription.go
package strand

func ToRNA(dna string) string {

	output := ""

	for i := 0; i < len(dna); i++{
		switch dna[i] {

		case 65:
			output += "U"
		case 67:
			output += "G"
		case 71:
			output += "C"
		case 84:
			output += "A"

		}
	}


	return output
}

